### PR TITLE
Rust: use brewed glibc if available to avoid GLIBC_XX not found errors on old systems

### PR DESCRIPTION
**Pull request by carlorosati**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Address issue #1837

### CODE REVIEW

File: Formula/rust.rb [hunk 1]
```diff
@@ -88,6 +88,11 @@ def install
 
     resource("cargobootstrap").stage do
       system "./install.sh", "--prefix=#{buildpath}/cargobootstrap"
+      if OS.linux? and Formula["glibc"].installed?
+        keg = Keg.new(prefix)
+        file = Pathname.new('#{buildpath}/cargobootstrap')
+        keg.change_rpath(file, Keg::PREFIX_PLACEHOLDER, HOMEBREW_PREFIX.to_s)
```
Comment (sjackman) [inline Formula/rust.rb]: Change this to (untested)
```ruby
Keg.new(prefix). keg.change_rpath(buildpath/"cargobootstrap",
    Keg::PREFIX_PLACEHOLDER, HOMEBREW_PREFIX)
```

**Comment by sjackman** [comment]
I haven't been able to get `rust` to build on CircleCI. See #1953 
With 13 threads it uses more than the available 4 GB of RAM on CircleCI.
With 12 threads it times out after 2 hours.
Any thoughts to either reduce the memory usage or reduce the build time of `rust`?

**Comment by carlorosati** [comment]
My first thought is to turn off optimization. I'll try to think more.

**Comment by carlorosati** [comment]
Try 

```
--enable-fast-make
 --disable-docs
--disable-optimize-tests
```

Also I'm not sure if ```--disable-dependency-tracking``` [1] will do anything but it's worth a try.

[1] https://www.gnu.org/software/automake/manual/html_node/Dependency-Tracking.html

**Comment by sjackman** [comment]
The upstream authors build Rust on TravisCI in 1h9m. See https://travis-ci.org/rust-lang/rust/jobs/211613672
Their build script: https://github.com/rust-lang/rust/blob/master/src/ci/run.sh

**Comment by carlorosati** [comment]
--enable-sccache                 invoke gcc/clang via sccache to reuse object files between builds

Sounds promising

**Comment by carlorosati** [comment]
reopening. new commit isn't showing.... help?

**Comment by carlorosati** [comment]
@sjackman This is still broken. Can you reopen this?

**Comment by sjackman** [comment]
We have a bottle for rust now. CircleCI increased our build time limit

**Comment by carlorosati** [comment]
Okay well that's good to hear, but it still is failing to build b/c of the newer libc on an older system. See the logs. https://gist.github.com/anonymous/aa59b3a242986338fafe8cac99e1df3b

**Comment by carlorosati** [comment]
Okay well that's good to hear, but it still is failing to build b/c of the newer libc on an older system. See the logs. https://gist.github.com/anonymous/aa59b3a242986338fafe8cac99e1df3b

**Comment by carlorosati** [comment]
Okay well that's good to hear! However, it still is fails to build without finding the newer libc on an older system. See the logs. https://gist.github.com/anonymous/aa59b3a242986338fafe8cac99e1df3b

**Comment by carlorosati** [comment]
Okay well that's good to hear! However, it still fails to build without finding the newer libc on an older system. See the logs. https://gist.github.com/anonymous/aa59b3a242986338fafe8cac99e1df3b

**Comment by sjackman** [comment]
```
   Compiling bootstrap v0.0.0 (file:///tmp/rust-20171108-25254-ancgse/rustc-1.21.0-src/src/bootstrap)
error: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /tmp/rust-20171108-25254-ancgse/rustc-1.21.0-src/build/bootstrap/debug/deps/libserde_derive-d72f95c90d043394.so)
   --> src/bootstrap/lib.rs:123:1
    |
123 | extern crate serde_derive;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Could not compile `bootstrap`.
```
https://gist.github.com/anonymous/aa59b3a242986338fafe8cac99e1df3b#file-02-make-L55

**Comment by sjackman** [comment]
I don't have time to work on this issue myself, sorry. A PR would be most welcome if you're able to find a fix.

Consider installing Linuxbrew in `/home/linuxbrew/.linuxbrew/` if possible so that you can use precompiled binary packages (known as bottles) for non-relocatable formula like `rust`.
Another possible workaround for you is `brew install --force-bottle rust`, but no promises.

The precompiled binary bottles of non-relocatable bottles can only be used if you install in `/home/linuxbrew/.linuxbrew`, otherwise they have to be built from source. See the documentation below, but wherever it reads `/usr/local`, which is for macOS, read instead `/home/linuxbrew/.linuxbrew`.
+ http://docs.brew.sh/Installation.html#alternative-installs
+ http://docs.brew.sh/FAQ.html#why-do-you-compile-everything
+ http://docs.brew.sh/Bottles.html#cellar-cellar

**Pull request closed without merging**